src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -0,0 +1,260 @@
+                arity: 0,
+                LookupOptions.MustBeInstance | LookupOptions.MustBeInvocableIfMember,
+                this,
+                this.ContainingType,
+                diagnose: false,
+                ref useSiteDiagnostics,
+                ConsList<TypeSymbol>.Empty);
+
+            MethodSymbol? withMethod = null;
+            if (lookupResult.IsSingleViable &&
+                lookupResult.SingleSymbolOrDefault is MethodSymbol m &&
+                m.ContainingType.Equals(receiverType, TypeCompareKind.ConsiderEverything))
+            {
+                withMethod = m;
+            }
+            else if (lookupResult.IsMultiViable)
+            {
+                // If there are multiple With methods, exclude overrides and
+                // look for a single remaining one
+                foreach (var member in lookupResult.Symbols)
+                {
+                    if (member is MethodSymbol { IsOverride: false } method &&
+                        method.ContainingType.Equals(receiverType, TypeCompareKind.ConsiderEverything))
+                    {
+                        if (withMethod is null)
+                        {
+                            withMethod = method;
+                        }
+                        else
+                        {
+                            withMethod = null;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            lookupResult.Clear();
+            useSiteDiagnostics = null;
+            ImmutableArray<Symbol?> withMembers;
+
+            if (withMethod is null)
+            {
+                diagnostics.Add(ErrorCode.ERR_NoSingleWithMethod, syntax.Receiver.Location, receiverType);
+                withMembers = default;
+            }
+            else
+            {
+                // Check return type
+                if (!withMethod.ReturnType.IsEqualToOrDerivedFrom(
+                    receiverType,
+                    TypeCompareKind.ConsiderEverything,
+                    ref useSiteDiagnostics))
+                {
+                    diagnostics.Add(
+                        ErrorCode.ERR_ContainingTypeMustDeriveFromWithReturnType,
+                        syntax.Receiver.Location,
+                        receiverType,
+                        withMethod.ReturnType);
+                }
+
+                useSiteDiagnostics = null;
+
+                // Build WithMethod member list
+                var matchingMembers = ArrayBuilder<Symbol?>.GetInstance(withMethod.ParameterCount);
+                foreach (var p in withMethod.Parameters)
+                {
+                    this.LookupMembersInType(
+                        lookupResult,
+                        receiverType,
+                        p.Name,
+                        arity: 0,
+                        basesBeingResolved: null,
+                        options: LookupOptions.MustBeInstance,
```

Comment:
>MustBeInstance [](start = 47, length = 14)

I don't think this should affect the *lookup*, only the result.  This flag is only used currently for handling compiler-generated symbols while translating query expressions, never for symbols appearing in source.  I think you need a separate error if the lookup yields a static member.

Reply:
Interesting. OK, I'll adjust the lookup.